services/src/main/java/edu/unc/lib/dl/cdr/services/processing/RunEnhancementsService.java

```diff
@@ -0,0 +1,132 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.cdr.services.processing;
+
+/**
+ * @author lfarrell
+ */
+import static edu.unc.lib.dl.xml.JDOMNamespaceUtil.ATOM_NS;
+import static edu.unc.lib.dl.xml.JDOMNamespaceUtil.CDR_MESSAGE_NS;
+
+import io.dropwizard.metrics5.Timer;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.HashMap;
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.unc.lib.dl.acl.service.AccessControlService;
+import edu.unc.lib.dl.acl.util.AgentPrincipals;
+import edu.unc.lib.dl.acl.util.Permission;
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.metrics.TimerFactory;
+import edu.unc.lib.dl.model.DatastreamPids;
+import edu.unc.lib.dl.search.solr.model.BriefObjectMetadata;
+import edu.unc.lib.dl.search.solr.model.Datastream;
+import edu.unc.lib.dl.search.solr.model.SearchRequest;
+import edu.unc.lib.dl.search.solr.model.SearchResultResponse;
+import edu.unc.lib.dl.search.solr.model.SearchState;
+import edu.unc.lib.dl.search.solr.model.SimpleIdRequest;
+import edu.unc.lib.dl.services.AbstractMessageSender;
+import edu.unc.lib.dl.ui.service.SolrQueryLayerService;
+
+public class RunEnhancementsService extends AbstractMessageSender {
+    private static final Logger LOG = LoggerFactory.getLogger(RunEnhancementsService.class);
+    private static final Timer timer = TimerFactory.createTimerForClass(RunEnhancementsService.class);
+
+    private AccessControlService aclService;
+
+    @Autowired
+    protected SolrQueryLayerService queryLayer;
+    private Datastream recordInfo;
+
+    public void run(AgentPrincipals agent, ArrayList<HashMap> objectPids, Boolean force) {
+        try (Timer.Context context = timer.time()) {
+            for (HashMap objectPid : objectPids) {
+                String uuid = (String) objectPid.get("pid");
+                PID pid = PIDs.get(uuid);
+
+                aclService.assertHasAccess("User does not have permission to run enhancements",
+                        pid, agent.getPrincipals(), Permission.runEnhancements);
+
+                LOG.debug("sending solr update message for {} of type {}", pid, "runEnhancements");
+
+                String objectType = (String) objectPid.get("objectType");
+
+                if (!objectType.equals("File")) {
+                    SearchRequest searchRequest = new SearchRequest();
+                    searchRequest.setAccessGroups(agent.getPrincipals());
+                    searchRequest.setSearchState(new SearchState());
```

Comment:
Since all of our enhancements are for files at the moment, you could limit the search to just return file objects. You can do that by adding a facet filter, something like:
`searchState.getFacets().put(SearchFieldKeys.RESOURCE_TYPE.name(),
                    ResourceType.File.name());`